### PR TITLE
feature/Fully cover commands with UT 2

### PR DESCRIPTION
**Pull request by okozlovlux**
- fully covered some commands with UT

Related to [APPLINK-28070](https://adc.luxoft.com/jira/browse/APPLINK-28070)

### CODE REVIEW

**Comment by okozlovlux** [comment]
@OHerasym @dtrunov @LevchenkoS @wolfylambova22 @pvvasilev @Kozoriz please review

File: src/components/application_manager/src/commands/mobile/on_vehicle_data_notification.cc [hunk 1]
```diff
@@ -68,10 +68,6 @@ void OnVehicleDataNotification::Run() {
 
       for (; applications.end() != app_it; ++app_it) {
         const ApplicationSharedPtr app = *app_it;
-        if (!app) {
-          SDL_ERROR("NULL pointer");
-          continue;
-        }
```
Comment (LevchenkoS) [inline src/components/application_manager/src/commands/mobile/on_vehicle_data_notification.cc]: @okozlovlux, perhaps, we should left, at least, `continue`.

File: src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc [hunk 2]
```diff
@@ -216,7 +218,10 @@ typedef Types<commands::NaviIsReadyRequest,
               commands::VRGetSupportedLanguagesRequest,
               commands::VRGetLanguageRequest,
               commands::VRPerformInteractionRequest,
-              commands::VRIsReadyRequest> RequestCommandsList2;
+              commands::VRIsReadyRequest,
+              commands::AllowAppRequest,
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc]: @okozlovlux, This one already included to `RequestCommandsList`

File: src/components/application_manager/test/commands/mobile/on_vehicle_data_notification_test.cc [hunk 3]
```diff
@@ -118,16 +118,17 @@ TEST_F(OnVehicleDataNotificationTest,
...
-  const int kFuelLevel = 100;
+  const uint32_t fuel_level = 100u;
   (*command_msg_)[am::strings::msg_params][am::strings::fuel_level] =
-      kFuelLevel;
+      fuel_level;
 
   MockAppPtr mock_app(CreateMockApp());
   std::vector<ApplicationSharedPtr> applications;
   applications.push_back(mock_app);
+  applications.push_back(mock_app);
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/on_vehicle_data_notification_test.cc]: @okozlovlux, why are you push this one twice?

File: src/components/application_manager/test/commands/hmi/activate_app_request_test.cc [hunk 4]
```diff
@@ -53,6 +53,13 @@ using am::commands::CommandImpl;
 
 typedef ::utils::SharedPtr<ActivateAppRequest> ActivateAppRequestPtr;
 
+MATCHER_P(CheckMessage, level, "") {
+  return level ==
+         static_cast<mobile_apis::HMILevel::eType>(
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/hmi/activate_app_request_test.cc]: @okozlovlux, perhaps, we also should check a result code too.

File: src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc (hunk 2 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc]: @LevchenkoS , no. it is not

Comment (LevchenkoS) [inline src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc]: @okozlovlux, sorry, my bad, I have been confused by `AllowAllAppsRequest`

File: src/components/application_manager/test/commands/mobile/on_vehicle_data_notification_test.cc (hunk 3 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/on_vehicle_data_notification_test.cc]: @LevchenkoS , for passing all code in command.

File: src/components/application_manager/test/commands/hmi/activate_app_request_test.cc (hunk 4 above)
Comment (okozlovlux) [inline src/components/application_manager/test/commands/hmi/activate_app_request_test.cc]: @LevchenkoS , there is no any checks of `result_code` in  this command

File: src/components/application_manager/src/commands/mobile/on_vehicle_data_notification.cc (hunk 1 above)
Comment (okozlovlux) [inline src/components/application_manager/src/commands/mobile/on_vehicle_data_notification.cc]: @LevchenkoS , fixed

**Comment by okozlovlux** [comment]
@AGritsevich please review

**Comment by okozlovlux** [comment]
@dtrunov @wolfylambova22 @Kozoriz please review

**Pull request merged**